File: src/main/java/eu/codetopic/utils/debug/items/ConnectivityDebugItem.java
/*
 * Copyright 2017 Jiří Kuchyňka (Anty)
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package eu.codetopic.utils.debug.items;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.TextView;

import eu.codetopic.utils.AndroidUtils;
import eu.codetopic.utils.NetworkManager;
import eu.codetopic.utils.R;
import eu.codetopic.utils.ui.container.items.custom.CardViewWrapper;
import eu.codetopic.utils.ui.container.items.custom.CustomItem;
import eu.codetopic.utils.ui.container.items.custom.CustomItemWrapper;

public class ConnectivityDebugItem extends CustomItem {

    private static final String LOG_TAG = "ConnectivityDebugItem";

    @Override
    protected void onBindViewHolder(ViewHolder holder, int itemPosition) {
        final TextView resultText = (TextView) holder.itemView.findViewById(R.id.resultText);
        holder.itemView.findViewById(R.id.checkButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                resultText.setVisibility(View.VISIBLE);
                resultText.setText(AndroidUtils.getFormattedText(resultText.getContext(),
                        R.string.debug_item_connectivity_info_result_text,
                        Boolean.toString(NetworkManager.isConnected(NetworkManager.NetworkType.ANY)),
                        Boolean.toString(NetworkManager.isConnected(NetworkManager.NetworkType.WIFI)),
                        Boolean.toString(NetworkManager.isConnected(NetworkManager.NetworkType.MOBILE))));
            }
        });
    }

    @Override
    public int getItemLayoutResId(Context context) {
        return R.layout.debug_item_connectivity;
    }

    @NonNull
    @Override
    protected CustomItemWrapper[] getWrappers(Context context) {
        return CardViewWrapper.WRAPPER;
    }
}

File: src/main/java/eu/codetopic/utils/debug/items/ConnectivityDebugItem.kt
/*
 * utils
 * Copyright (C)   2017  anty
 *
 * This program is free  software: you can redistribute it and/or modify
 * it under the terms  of the GNU General Public License as published by
 * the Free Software  Foundation, either version 3 of the License, or
 * (at your option) any  later version.
 *
 * This program is distributed in the hope that it  will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied  warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
 * GNU General Public License for more details.
 *
 * You  should have received a copy of the GNU General Public License
 * along  with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package eu.codetopic.utils.debug.items

import android.content.Context
import android.view.View
import android.widget.TextView

import eu.codetopic.utils.AndroidExtensions.getFormattedText
import eu.codetopic.utils.network.NetworkManager
import eu.codetopic.utils.R
import eu.codetopic.utils.network.NetworkManager.NetworkType.*
import eu.codetopic.utils.ui.container.items.custom.CardViewWrapper
import eu.codetopic.utils.ui.container.items.custom.CustomItem
import eu.codetopic.utils.ui.container.items.custom.CustomItemWrapper

class ConnectivityDebugItem : CustomItem() {

    companion object {

        private const val LOG_TAG = "ConnectivityDebugItem"
    }

    override fun onBindViewHolder(holder: CustomItem.ViewHolder, itemPosition: Int) {
        val resultText = holder.itemView.findViewById<TextView>(R.id.resultText)

        holder.itemView.findViewById<View>(R.id.checkButton).setOnClickListener {
            resultText.visibility = View.VISIBLE
            resultText.text = resultText.context
                    .getFormattedText(R.string.debug_item_connectivity_info_result_text,
                            NetworkManager.isConnected(ANY).toString(),
                            NetworkManager.isConnected(WIFI).toString(),
                            NetworkManager.isConnected(MOBILE).toString())
        }
    }

    override fun getItemLayoutResId(context: Context) = R.layout.item_debug_connectivity

    override fun getWrappers(context: Context): Array<CustomItemWrapper> = CardViewWrapper.WRAPPER
}
